cmake/compilerFlags.cmake

```diff
@@ -20,17 +20,24 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
         endif()
     endif()
 
-    if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
+    if (COMPILER_IS_GCC)
+        add_compile_options(-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS)
+        if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0 )
+            add_compile_options(-fstack-clash-protection -fcf-protection)
+        endif()
+    endif()
 
+    if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
         if(BUILD_WITH_COVERAGE)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g ")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftest-coverage")
+            # Note: We tried to use here add_compile_options but we got linker errors on Travis-CI
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage -fprofile-arcs -ftest-coverage")

```

Comment:
`--coverage` implies the latter two: https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#Instrumentation-Options